invenio_drafts_resources/services/records/components/base.py

```diff
@@ -186,6 +128,61 @@ def new_version(self, identity, draft=None, record=None):
         record_files = self.get_record_files(record)
         draft_files.enabled = record_files.enabled
 
+    def _assign_files_enabled(self, identity, data, draft, errors):
+        """Assign files.enabled."""
+        enabled = data.get(self.files_data_key, {}).get(
+            "enabled", self.service.config.default_files_enabled
+        )
+        if self._check_can_manage(identity, data, draft, errors, enabled):
+            try:
+                self.assign_files_enabled(draft, enabled)
+            except ValidationError as e:
+                errors.append(
+                    {"field": f"{self.files_data_key}.enabled", "messages": e.messages}
+                )
+                return  # exit early
+        else:
+            return  # exit early
+
+    def _check_can_manage(self, identity, data, draft, errors, enabled):
+        """Check whether identity can manage files."""
+        files = self.get_record_files(draft)
+
+        if files.enabled != enabled:
```

Comment:
minor: for readability it would help to add early returns. this would reduce the indentation levels